common/pb/response_ban_history.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto2";
+import "response.proto";
+
+message Response_BanHistory{
+    extend Response {
+        optional Response_BanHistory ext = 1012;
+    }
+    repeated string admin_id = 1;
+    repeated string admin_name = 2;
+    repeated string ban_time = 3;
+    repeated string ban_length = 4;
+    repeated string ban_reason = 5;
+    repeated string visible_reason = 6;
```

Comment:
I don't think we have to say that ServerInfo_Foo is only for things live in memory. It's info from the server. Even with your example of ServerInfo_User, that's a case where only working on online users is an accidental limitation rather than design goal.

Regardless, the point is to have the ban data inside a single logical unit. Keeping related data in unrelated arrays joined only by accidently sharing an index is not a good pattern to work with. It's error prone and makes the logic harder to follow.